common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -388,6 +388,57 @@ public static CharSequence escapeCsv(CharSequence value) {
     }
 
     /**
+     * Unescapes the specified escaped CSV field, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     *
+     * @param value The escaped CSV field which will be unescaped according to
+     *              <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>
+     * @return {@link CharSequence} the unescaped value if necessary, or the value unchanged
+     */
+    public static CharSequence unescapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(value.charAt(0)) && isDoubleQuote(value.charAt(last)) && length != 1;
+        StringBuilder unescaped = quoted ? InternalThreadLocalMap.get().stringBuilder() : null;
+        for (int i = 0; i < length; i++) {
+            char current = value.charAt(i);
+            switch (current) {
+                case DOUBLE_QUOTE:
+                    if (i == 0 || i == last) {
+                        if (length == 1) {
+                            // value is "\""
+                            throw new IllegalArgumentException("Invalid escaped csv field: " + value);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Invalid escaped csv field: " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)